src/hash.cr

```diff
@@ -272,6 +272,26 @@ class Hash(K, V)
     self
   end
 
+  def select(&block : K, V -> U)
+    inject({} of K => V) do |memo, k, v|
+      yield(k, v) ? memo.merge!({k => v}) : memo
```

Comment:
sure, `Ruby` has this defined on `Enumerable` does it make sense to put it there and have `Hash` inherit from it?